packages/gatsby-transformer-video/src/ffmpeg.js

```diff
@@ -0,0 +1,431 @@
+import crypto from "crypto"
+import http from "http"
+import { tmpdir } from "os"
+import { resolve, parse } from "path"
+
+import { pathExists, createWriteStream, ensureDir, unlink } from "fs-extra"
+import ffmpeg from "fluent-ffmpeg"
+import imagemin from "imagemin"
+import imageminGiflossy from "imagemin-giflossy"
+import PQueue from "p-queue"
+
+export default class FFMPEG {
+  constructor({ cacheDir, rootDir, ffmpegPath, ffprobePath }) {
+    this.queue = new PQueue({ concurrency: 1 })
+    this.cacheDir = cacheDir
+    this.rootDir = rootDir
+
+    if (ffmpegPath) {
+      ffmpeg.setFfmpegPath(ffmpegPath)
+    }
+    if (ffprobePath) {
+      ffmpeg.setFfprobePath(ffprobePath)
+    }
+  }
+
+  cacheContentfulVideo = async ({
+    id,
+    contentful_id: cid,
+    file: { url, fileName },
+    internal: { contentDigest },
+  }) => {
+    await ensureDir(this.cacheDir)
+
+    const { name, ext } = parse(fileName)
+    const absolutePath = resolve(
+      this.cacheDir,
+      `${name}-${contentDigest}${ext}`
+    )
+
+    const alreadyExists = await pathExists(absolutePath)
+
+    if (!alreadyExists) {
+      const previewUrl = `http:${url}`
+
+      console.log(`Downloading ${fileName} (${cid || id}) from ${previewUrl}`)
+
+      try {
+        await new Promise((resolve, reject) => {
+          const file = createWriteStream(absolutePath)
+          http
+            .get(previewUrl, function(response) {
+              response.pipe(file)
+              file.on(`finish`, function() {
+                file.close(resolve)
+              })
+            })
+            .on(`error`, function(err) {
+              reject(err)
+            })
+        })
+      } catch (err) {
+        await unlink(absolutePath)
+        throw err
+      }
+
+      console.log(`Finished downloading ${fileName} (${cid || id})`)
+    }
+
+    return absolutePath
+  }
+
+  analyzeVideo = async ({ video, fieldArgs, type }) => {
+    let path
+
+    if (type === `File`) {
+      path = video.absolutePath
+    }
+
+    if (type === `ContentfulAsset`) {
+      path = await this.queue.add(() => this.cacheContentfulVideo(video))
+    }
+
+    if (!path) {
+      throw new Error(
+        `Unable to extract asset file path for ${type} (${video.id})`
+      )
+    }
+
+    const optionsHash = crypto
+      .createHash(`md5`)
+      .update(JSON.stringify(fieldArgs))
+      .digest(`hex`)
```

Comment:
new code style suggestion from:

#8805 Issue: feat: update gatsby plugins to use new createContentDigest helper

```suggestion
    const optionsHash = createContentDigest(fieldArgs)
```
and add to imports:
```javascript
import { createContentDigest } from "gatsby-core-utils"
```

Reply:
cool! will do! thx

Reply:
done :)